diff -up aide-0.14/src/compare_db.c.reportattrsegv aide-0.14/src/compare_db.c
--- aide-0.14/src/compare_db.c.reportattrsegv	2014-07-16 13:43:46.184167375 +0200
+++ aide-0.14/src/compare_db.c	2014-07-16 13:44:08.490732759 +0200
@@ -607,7 +607,7 @@ void print_long_changes(const char* name
   snprintf(nline,part_len,"%lu",new);
 #endif
   if (justnew) {
-    error(2,(char*)entry_format,name,nline);
+    error(2,(char*)entry_format_justnew,name,' ',nline);
   } else {
     error(2,(char*)entry_format,name,oline,nline);
   }
@@ -620,7 +620,7 @@ void print_string_changes(const char* na
   }
   snprintf(nline,part_len,"%s",new);
   if (justnew) {
-    error(2,(char*)entry_format,name,nline); 
+    error(2,(char*)entry_format_justnew,name,' ',nline);
   } else {
     error(2,(char*)entry_format,name,oline,nline); 
   }
