diff -up aide-0.14/src/aide.c.fipsfix aide-0.14/src/aide.c
--- aide-0.14/src/aide.c.fipsfix	2012-04-03 11:26:01.361261991 +0200
+++ aide-0.14/src/aide.c	2012-04-05 13:35:55.897230124 +0200
@@ -454,7 +454,6 @@ void setdefaults_after_config()
   }
 }
 
-
 int main(int argc,char**argv)
 {
   int errorno=0;
@@ -542,6 +541,24 @@ int main(int argc,char**argv)
     }
   }
 #endif
+#if WITH_GCRYPT
+  error(255,"Gcrypt library initialization\n");
+  /* 
+   *  Initialize libgcrypt as per
+   *  http://www.gnupg.org/documentation/manuals/gcrypt/Initializing-the-library.html
+   *
+   *  
+   */
+  gcry_control(GCRYCTL_SET_ENFORCED_FIPS_FLAG, 0);
+  gcry_control(GCRYCTL_INIT_SECMEM, 1);
+
+  if(!gcry_check_version(GCRYPT_VERSION)) {
+      error(0,"libgcrypt version mismatch\n");
+      exit(VERSION_MISMATCH_ERROR);
+  }
+
+  gcry_control(GCRYCTL_INITIALIZATION_FINISHED, 0);
+#endif /* WITH_GCRYPT */
   conf->use_initial_errorsto=0;
   if (!conf->config_check) {
     int exitcode=0;
@@ -620,6 +637,9 @@ int main(int argc,char**argv)
     }
 #endif
   }
+#ifdef WITH_GCRYPT
+  gcry_control(GCRYCTL_TERM_SECMEM, 0);
+#endif /* WITH_GCRYPT */
   return RETOK;
 }
 const char* aide_key_3=CONFHMACKEY_03;
diff -up aide-0.14/src/md.c.fipsfix aide-0.14/src/md.c
--- aide-0.14/src/md.c.fipsfix	2012-04-03 11:26:01.354261970 +0200
+++ aide-0.14/src/md.c	2012-04-03 11:26:01.368262013 +0200
@@ -200,17 +200,7 @@ int init_md(struct md_container* md) {
   }
 #endif 
 #ifdef WITH_GCRYPT
-  error(255,"Gcrypt library initialization\n");
-  	/* 
-  	  Initialize libgcrypt as per
-  	  http://www.gnupg.org/documentation/manuals/gcrypt/Initializing-the-library.html
-  	 */
-  	if(!gcry_check_version(GCRYPT_VERSION)) {
-		error(0,"libgcrypt version mismatch\n");
-		exit(VERSION_MISMATCH_ERROR);
-	}
-	gcry_control(GCRYCTL_INITIALIZATION_FINISHED, 0);
-	if(gcry_md_open(&md->mdh,0,0)!=GPG_ERR_NO_ERROR){
+	if(gcry_md_open(&md->mdh,0,GCRY_MD_FLAG_SECURE)!=GPG_ERR_NO_ERROR){
 		error(0,"gcrypt_md_open failed\n");
 		exit(IO_ERROR);
 	}
@@ -302,7 +292,8 @@ int close_md(struct md_container* md) {
   /*.    There might be more hashes in the library. Add those here..   */
   
   gcry_md_reset(md->mdh);
-#endif  
+  gcry_md_close(md->mdh);
+#endif
 
 #ifdef WITH_MHASH
 #define get_mhash_hash(b,c) \
diff -up aide-0.14/src/util.c.fipsfix aide-0.14/src/util.c
--- aide-0.14/src/util.c.fipsfix	2012-04-03 11:26:01.359261985 +0200
+++ aide-0.14/src/util.c	2012-04-03 11:26:01.368262013 +0200
@@ -493,28 +493,5 @@ int syslog_facility_lookup(char *s)
 	return(AIDE_SYSLOG_FACILITY);
 }
 
-/* We need these dummy stubs to fool the linker into believing that
-   we do not need them at link time */
-
-void* dlopen(char*filename,int flag)
-{
-  return NULL;
-}
-
-void* dlsym(void*handle,char*symbol)
-{
-  return NULL;
-}
-
-void* dlclose(void*handle)
-{
-  return NULL;
-}
-
-const char* dlerror(void)
-{
-  return NULL;
-}
-
 const char* aide_key_2=CONFHMACKEY_02;
 const char* db_key_2=DBHMACKEY_02;
