diff -up ./src/gen_list.c.fix ./src/gen_list.c
--- ./src/gen_list.c.fix	2016-01-18 13:31:53.368352078 +0100
+++ ./src/gen_list.c	2016-01-18 13:35:15.762603650 +0100
@@ -61,6 +61,7 @@ void no_hash(db_line* line);
 char* strrxtok(char* rx)
 {
   char*p=NULL;
+  char*t=NULL;
   size_t i=0;
 
   /* The following code assumes that the first character is a slash */
@@ -83,8 +84,10 @@ char* strrxtok(char* rx)
 	i=strlen(p);
 	break;
       case '\\':
-	strcpy(p+i,p+i+1);
-	i++;
+	t=strdup(p);
+	strcpy(p+i,t+i+1);
+	free(t);
+	t=NULL;
 	break;
       default:
 	break;
